### PR TITLE
New envs api and env plugin

### PR DESCRIPTION
**Pull request by GiladShoham**
## Proposed Changes

- new env plugin (.bit-env)
- do not dedup dev server by peer dependencies in case rootComponents is enabled
- new XXX-env-type interfaces
- when compiling group of components, compile all envs first
- compute dev files from env.jsonc file
- all aspects to register DevPatternDescriptor which contain a name of the pattern - this name is used by env.jsonc as the key
- add names to pattern registered by docs, composition and tester aspects
- make worker.declareWorker sync op
- calculate env policy from env.jsonc file
- fix pnpm's publicHoistPattern to handle eslint plugins correctly
- new ServiceHandlerContext class
- new ServiceHandlerFactory interface
- change ServiceHandler interface
- get components templates from env (getGeneratorTemplates)
- get starters templates from env (getGeneratorStarters)
- new StarterList class 
- new TemplateList class 
- add `--aspect` flag to bit templates command, to filter templates in result
- fallback to core templates if template not found in manually registered template
- add `aspectFilePath` (*.aspect.ts|js) prop to `AspectDefinition` type
- add `OnAspectLoadErrorHandler` to plugins class
- try to re-load plugins that were failed to load after re-compile/install
- new Pipeline class
- new PackageGenerator class
- write preview root file into dist folder of the component instead of the root of the component
- move error boundaries outside of the composition app component
- on preview root file, import directly from the aspect file (.aspect.ts|js) instead of the aspect root folder
- only bundle host dependencies which directly declared like this (do not auto add any env's peers into the list) 
- remove @babel/runtime and core-js from deps of any component uses the react env (or envs that extends react)

### CODE REVIEW

File: scopes/component/ui/version-dropdown/version-dropdown.module.scss [hunk 1]
```diff
@@ -1,4 +1,7 @@
+@import '~@teambit/ui-foundation.ui.constants.z-indexes/z-indexes.module.scss';
```
Comment (luvkapur) [inline scopes/component/ui/version-dropdown/version-dropdown.module.scss:1]: can be removed from this PR

Will be fixed in #6714 
https://github.com/teambit/bit/pull/6714/files#diff-3fa370f6626e7ecad761bf8adfafad3e23458aaf5146c8bcb382440f7379b7e8

File: scopes/lanes/ui/inputs/lane-selector/lane-selector.module.scss [hunk 2]
```diff
@@ -1,6 +1,7 @@
 @import '~@teambit/ui-foundation.ui.constants.z-indexes/z-indexes.module.scss';
```
Comment (luvkapur) [inline scopes/lanes/ui/inputs/lane-selector/lane-selector.module.scss:1]: can be removed from this PR 
Will be fixed in #6714 
https://github.com/teambit/bit/pull/6714/files#diff-b8aa2ebf97d23e10dd1e64a8251dd683dd77d11f42260c7ebec6fea3547c1405

File: scopes/lanes/ui/navigation/lane-switcher/lane-switcher.module.scss [hunk 3]
```diff
@@ -13,6 +13,8 @@
 .laneGalleryIcon {
```
Comment (luvkapur) [inline scopes/lanes/ui/navigation/lane-switcher/lane-switcher.module.scss:13]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-0ec3f7734dff24c4d9ce2b21690bea773a5543a36dc7f3786025c98424e7dc6e

File: scopes/lanes/ui/navigation/lane-switcher/lane-switcher.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import React, { HTMLAttributes } from 'react';
```
Comment (luvkapur) [inline scopes/lanes/ui/navigation/lane-switcher/lane-switcher.tsx:1]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-e764c0dcda6133bbea98ddc9602826023931e423a209ad2bc3c21c87af3be9ab

File: scopes/ui-foundation/use-box/dropdown/dropdown.module.scss [hunk 5]
```diff
@@ -1,5 +1,9 @@
 @import '~@teambit/base-ui.layout.breakpoints/_breakpoints.scss';
```
Comment (luvkapur) [inline scopes/ui-foundation/use-box/dropdown/dropdown.module.scss:1]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-c6ad0b4f888ffd34e5951ae088072825f01723ad2e8388179ccc9b756e948b88

File: scopes/ui-foundation/use-box/menu/menu.module.scss [hunk 6]
```diff
@@ -1,8 +1,14 @@
+@import '~@teambit/ui-foundation.ui.constants.z-indexes/z-indexes.module.scss';
```
Comment (luvkapur) [inline scopes/ui-foundation/use-box/menu/menu.module.scss:1]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-e91ae23f82a550ba9ccc73a109410e51b516d491e817bf7e7a72387bfbaa9dc6

File: scopes/ui-foundation/use-box/menu/menu.tsx [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import React, { useState, useMemo, ReactNode } from 'react';
```
Comment (luvkapur) [inline scopes/ui-foundation/use-box/menu/menu.tsx:1]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-da180d74b8fa499c7a01ddcb24322e320f17c1d02fd29d037f1a2d72d5251b45

File: scopes/workspace/workspace/workspace.ui.runtime.tsx [hunk 8]
```diff
@@ -235,7 +235,7 @@ export class WorkspaceUI {
       component: function Gallery() {
```
Comment (luvkapur) [inline scopes/workspace/workspace/workspace.ui.runtime.tsx:235]: can be removed from this PR 
Will be fixed in #6714 

https://github.com/teambit/bit/pull/6714/files#diff-dd0fc6034d2ab3a73eb49449133831c5bba5463feabd33b3ef40944653037576

File: scopes/compositions/ui/composition-card/composition-card.tsx [hunk 9]
```diff
@@ -23,11 +21,9 @@ export function CompositionCard({ Composition, name, link, className, ...rest }:
 
   return (
     <Card elevation="low" className={classNames(className, styles.compositionCard)} {...rest}>
```
Comment (luvkapur) [inline scopes/compositions/ui/composition-card/composition-card.tsx:23]: Can be removed from this PR

This is in backlog to implement. We will be re designing the error fallback screens.

**Review by guysaar223 — APPROVED** [review]

**Pull request merged**